tests/test_transform.py

```diff
@@ -0,0 +1,171 @@
+                        local_variables={"n": "x"},
+                    )
+                },
+                type=None,
+                linked_params={"z": [("ccry_gate", "x")], "num": [("ccry_gate", "num")]},
+                ports={
+                    "in": {"name": "in", "direction": "input", "size": "z"},
+                    "out": {"name": "out", "direction": "output", "size": "z"},
+                },
+                connections=[
+                    {"source": "in", "target": "ccry_gate.in"},
+                    {"source": "ccry_gate.out", "target": "out"},
+                ],
+            ),
+        ),
+        (  # Example using aggregation dict values with parameters to approximate single-qubit Z-rotations
+            # using optimal ancilla-free Clifford+T circuits, as detailed in (arXiv:1403.2975).
+            {"arbitrary_z": {"T_gates": "3*log_2(1/epsilon) + O(log(log(1/epsilon)))"}},
+            generate_test(
+                arbitrary_z,
+                ["z", "num"],
+                [{"source": "z", "targets": ["arbitrary_z.x"]}, {"source": "num", "targets": ["arbitrary_z.num"]}],
+            ),
+            Routine(
+                name="test_qref",
+                input_params=["z", "num"],
+                children={
+                    "arbitrary_z": Routine(
+                        name="arbitrary_z",
+                        type=None,
+                        input_params=["x", "num"],
+                        ports={
+                            "in": {"name": "in", "direction": "input", "size": "n"},
+                            "out": {"name": "out", "direction": "output", "size": "n"},
+                        },
+                        resources={
+                            "T_gates": {
+                                "name": "T_gates",
+                                "type": "additive",
+                                "value": "ceiling(5*num/2)*(3*log_2(1/epsilon) + O(log(log(1/epsilon))))",
+                            },
+                        },
+                        local_variables={"n": "x"},
+                    )
+                },
+                type=None,
+                linked_params={"z": [("arbitrary_z", "x")], "num": [("arbitrary_z", "num")]},
+                ports={
+                    "in": {"name": "in", "direction": "input", "size": "z"},
+                    "out": {"name": "out", "direction": "output", "size": "z"},
+                },
+                connections=[
+                    {"source": "in", "target": "arbitrary_z.in"},
+                    {"source": "arbitrary_z.out", "target": "out"},
+                ],
+            ),
+        ),
+    ],
+)
+def test_add_aggregated_resources(aggregation_dict, generate_test_fn, expected_output):
+    result = add_aggregated_resources(aggregation_dict, generate_test_fn)
+    compare_routines(result, expected_output)
+
+
+# Define the comparison function
+def compare_routines(routine, expected):
```

Comment:
[suggestion] This one also could be private. Plus, in this case the comment is unnecessary, it's pretty obvious what this function does from its name.